src/client/opamCliMain.ml

```diff
@@ -73,26 +85,26 @@ let rec preprocess_argv cli yes args =
           raise (InvalidCLI (Ok ocli))
       | _ -> raise (InvalidCLI (Error (Some arg)))
     in
-    preprocess_argv (Some version) yes args
+    preprocess_argv (Some version) yes confirm args
   | arg :: rest ->
     match OpamStd.String.cut_at arg '=' with
     | Some ("--cl", value)
```

Comment:
Handling for `--confirm-level` needed here too - at which point the `prefix_of` function saves us from code duplication!